src/en/robust-toolbox/ecs.md

```diff
@@ -110,26 +110,25 @@ As you can see, by using composition all our problems from before are solved nea
-The way you would do this in E/C would be to put all of that logic in the "amount" property on `StackComponent`, or maybe a method. Then you could `private` the actual "amount" number away. However the E/C architecture will not be accepted in this codebase, we will only use the ECS architecture going forward.
+But perhaps that's the solution? Mobs can have machine bits, and machines can have mob bits, so just create an abstract machine mob class! But at the same time, that wouldn't make much sense either! Most mobs will not need "power consumption features", and few machines will be player-controlled at all.
 
-See, putting any kind of logic in a component class doesn't make sense at all. If we think about how stuff in ECS is structured, it's like this:
-1. Our game world has a bunch of entities, the entities have components
-2. There are systems that operate on components
+To solve this difficult design decision, we would humbly propose an alternative design perspective. Instead of inheriting features, what if we composed them? Think of it as building legos instead of branching trees, but how would this work? Suppose you want an entity to have hands, add a `HandsComponent`. If you want to make it consume power, `PowerConsumerComponent` will help! If you want to make it player-controllable, add `MindComponent`. What have we created? An entity with hands, power, and can be controlled by a player. We've made a "cyborg mob" out of reusable, generic components! 
 
-Therefore, components should only have data in them, and no logic whatsoever. Systems should give the entities their behavior, when they have the appropriate components.
-I'm more saying like... Rather than have logic in components change stuff, we want entity systems to operate and modify components. So instead of having:
-`HandsComponent.Pickup(ItemComponent)`, you would have:
-`HandsSystem.Pickup(UserEntity, ItemEntity)`.
+Inheritance may be perfectly fine in nuanced circumstances, such as small, self-contained inheritance trees (see something like `SoundSpecifier` for example). However, at scale its benefits become detriments quickly. Especially when you have a game as large and complex as ss14.
 
+Now let's take a look at encapsulation. OOP associates data and behavior in the same class. Such classes only expose certain properties or methods to external parties via access modifiers. That is, fields such as `public`, `private` and `protected`. Encapsulation is good for cases such as our toolbax engine; which specifically needs to obscure/prohibit access to some data or methods. But for game and data logic, it doesn't make that much sense. For example, the `StackComponent` in SS14 has no private fields or properties and any owner is free to read or write the values as they please. However, this is not the recommended way to interact with stacks! 
```

Comment:
How would you respond to the question, "it sounds like SS14 is using access modifiers wrong by not hiding private internal data fields?" OOP is also about encapsulation. What is different between `StackSystem.Split()` vs a `stackInstance.Split()` method?